<launch>
  <arg name="use_imu_filter"         default="true"/>
  <arg name="raw_pointcloud_topic1"   default="/d455_r/depth/color/points"/>
  <arg name="raw_pointcloud_topic2"   default="/d455_l/depth/color/points"/>
  <arg name="camera_height"          default="1.0"/>
  <arg name="base_frame_id"          default="base_link"/>
  <arg name="right_frame_id"         default="d455_r_depth_optical_frame"/>
  <arg name="left_frame_id"          default="d455_l_depth_optical_frame"/>
  <arg name="max_sensing_range"      default="6.0"/>
  <arg name="grid_unit_size"         default="0.15"/>
  <arg name="grid_length"            default="15.0"/>
  <arg name="grid_width"             default="15.0"/>
  <arg name="grid_height"            default="1.0"/>
  
  <param name="/use_sim_time"        value="false"/>
  
 <group if="$(arg use_imu_filter)">
  <!-- Launch the Complementary filter node -->
  <node pkg="imu_complementary_filter" type="complementary_filter_node" name="complementary_filter_gain_node" output="screen">
    <remap from="input_raw_imu"                                  to = "/d455_r/imu" />
    <param name="do_bias_estimation"        type = "bool"        value = "true"/>
    <param name="do_adaptive_gain"          type = "bool"        value = "true"/>
    <param name="use_mag"                   type = "bool"        value = "false"/>
    <param name="gain_acc"                  type = "double"      value = "0.01"/>
    <param name="gain_mag"                  type = "double"      value = "0.01"/>
  </node>
 </group>
  
  <!-- Launch the ground plane segmentation node -->
  <node pkg="mux_seg" type="mux_seg" name="mux_seg" output="screen">
    <remap from="cloud_in_1"                                       to = "$(arg raw_pointcloud_topic1)" />
    <remap from="cloud_in_2"                                       to = "$(arg raw_pointcloud_topic2)" />
    
    <remap from="/range_in_right"                                  to = "/evo_mini_0/pointcloud" />
    <remap from="/range_in_left"                                   to = "/evo_mini_1/pointcloud" />
    <remap from="/lidar_in_right"                                  to = "/ydlidar_0/point_cloud" />
    <remap from="/lidar_in_left"                                   to = "/ydlidar_1/point_cloud" />
    
    <!--<param name="base_frame_id"             type = "string"      value = "base_optical_link" />-->
    
    <param name="right_camera_frame_id"     type = "string"      value = "$(arg right_frame_id)" />
    <param name="left_camera_frame_id"      type = "string"      value = "$(arg left_frame_id)" />
    <param name="right_range_frame_id"      type = "string"      value = "base_range_0" />
    <param name="left_range_frame_id"       type = "string"      value = "base_range_1" />
    <param name="right_lidar_frame_id"      type = "string"      value = "laser_frame_0" />
    <param name="left_lidar_frame_id"       type = "string"      value = "laser_frame_1" />
    
    <param name="sensor_height"             type = "double"      value = "$(arg camera_height)"/>
    <param name="num_iter"                  type = "int"         value = "5"/>
    <param name="num_lpr"                   type = "int"         value = "400"/>
    <param name="th_seeds"                  type = "double"      value = "0.5"/><!--1.2-->
    <param name="th_dist"                   type = "double"      value = "0.05"/><!--0.1-->
    <param name="th_ceil_"                  type = "double"      value = "$(arg grid_height)"/>
    <param name="th_box_"                   type = "double"      value = "$(arg max_sensing_range)"/>
    <param name="map_unit_size_"            type = "double"      value = "$(arg grid_unit_size)"/>
    <param name="radius_search_"            type = "double"      value = "0.15"/>
    <param name="in_radius_"                type = "int"         value = "25"/>
    <param name="std_th_"                   type = "double"      value = "0.1"/>
    <param name="mean_k_"                   type = "int"         value = "-15"/>
    <param name="alpha"                     type = "double"      value = "0.1"/>
    <param name="SVD_refinement"            type = "bool"        value = "false"/>
    <param name="dense"                     type = "bool"        value = "false"/>
    <param name="detect_neg_obstacle"       type = "bool"        value = "false"/>
    <param name="debug"                     type = "bool"        value = "false"/>
  </node>
  
 
  
</launch>
